src/runtime/pkg/resourcecontrol/cgroups.go

```diff
@@ -309,12 +315,14 @@ func (c *LinuxCgroup) AddProcess(pid int, subsystems ...string) error {
 	}
 }
 
-func (c *LinuxCgroup) AddThread(pid int, subsystems ...string) error {
+func (c *LinuxCgroup) AddThread(tid int, subsystems ...string) error {
 	switch cg := c.cgroup.(type) {
 	case cgroups.Cgroup:
-		return cg.AddTask(cgroups.Process{Pid: pid})
+		return cg.AddTask(cgroups.Process{Pid: tid})
 	case *cgroupsv2.Manager:
-		return cg.AddProc(uint64(pid))
+		// Currently containerd/cgroups does not provide the AddThread() method.
+		// To be replaced by the method in manager after https://github.com/containerd/cgroups/pull/243 is merged.
+		return ioutil.WriteFile(filepath.Join(unifiedMountpoint, c.ID(), "cgroup.threads"), []byte(strconv.FormatUint(uint64(tid), 10)), os.FileMode(0))
```

Comment:
`ioutil.WriteFile` is deprecated, please use `os.WriteFile` instead.

Reply:
Already used `os.WriteFile` instead. `runc_cgroups.WriteFile` only allows the parameter type to be `string` and cannot write to cgroup.threads.